data-products/src/main/scala/org/sunbird/analytics/archival/BaseArchivalJob.scala

```diff
@@ -69,46 +69,50 @@ trait BaseArchivalJob extends BaseReportsJob with IJob with ArchivalMetaDataStor
-    val archivalKeyspace = archivalRequest.keyspace.getOrElse(AppConf.getConfig("sunbird.courses.keyspace"))
+    val requestConfig = JSONUtils.deserialize[Request](JSONUtils.serialize(modelParams.getOrElse("request", Request).asInstanceOf[Map[String,AnyRef]]))
+    val archivalTable = requestConfig.archivalTable
+    val archivalKeyspace = requestConfig.keyspace.getOrElse(AppConf.getConfig("sunbird.courses.keyspace"))
 
-    val batchId: String = archivalRequest.batchId.getOrElse("")
-    val date: String  = archivalRequest.date.getOrElse("")
+    val batchId: String = requestConfig.batchId.getOrElse("")
+    val date: String  = requestConfig.date.getOrElse("")
     val mode: String = modelParams.getOrElse("mode","archive").asInstanceOf[String]
 
     println("modelParams: " + modelParams)
-    println("archival request: " + archivalRequest)
+    println("archival request: " + requestConfig)
     val archivalTableData: DataFrame = getArchivalData(archivalTable, archivalKeyspace,Option(batchId),Option(date))
     println("archivalTableData ")
     archivalTableData.show(false)
 
     mode.toLowerCase() match {
       case "archival" =>
-        archiveData(archivalTableData, archivalRequest)
+        archiveData(archivalTableData, requestConfig)
       case "delete" =>
-        deleteArchivedData(archivalTableData,archivalRequest)
+        deleteArchivedData(archivalTableData,requestConfig)
     }
   }
 
-  def archiveData(data: DataFrame, archivalRequest: Request)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): Unit = {
-    val requests = getRequests(jobId, archivalRequest.batchId)
+  def archiveData(data: DataFrame, requestConfig: Request)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): Unit = {
+    val requests = getRequests(jobId, requestConfig.batchId)
     println("requestLength: " + requests.length)
     try {
-      if(requests.length == 0) {
-        val dataDF = processArchival(data, archivalRequest)
-        dataDF.show(false)
-        val archiveBatchList = dataDF.groupBy(partitionCols.head, partitionCols.tail: _*).count().collect()
-        println("archiveBatchList: " + archiveBatchList.head)
-
-        val batchesToArchive: Map[String, Array[BatchPartition]] = archiveBatchList.map(f => BatchPartition(f.get(0).asInstanceOf[String], Period(f.get(1).asInstanceOf[Int], f.get(2).asInstanceOf[Int]))).groupBy(_.batchId)
-
-        val archivalStatus = archiveBatches(batchesToArchive, data, requests.head, archivalRequest)
-      } else {
-        for (request <- requests) yield {
-        //TO-DO: for each request
+      var dataDF = processArchival(data, requestConfig)
+      if(requests.length > 0) {
+        for (request <- requests) {
+          if (request.archival_status.equals("SUCCESS")) {
+            val request_data = JSONUtils.deserialize[Map[String, AnyRef]](request.request_data)
+            dataDF = dataDF.filter(
+              col("batch_id").equalTo(request.batch_id) &&
```

Comment:
Please do filter on course id as well

Reply:
We are already doing a filter when we fetch the archival table records.